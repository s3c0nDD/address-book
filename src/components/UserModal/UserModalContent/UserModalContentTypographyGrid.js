import React from 'react';
import PropTypes from 'prop-types';

import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

/**
 * User modal content typography component
 * @param {object} props component's props
 * @param {string} props.text text to present
 */
const UserModalContentTypography = ({ text }) => (
  <Typography
    variant="body2"
    color="textSecondary"
    component="p"
  >
    {text}
  </Typography>
);

/**
 * User modal content typography grid component
 * @param {object} props component's props
 * @param {function} props.IconComponent icon component to present with text
 * @param {function} props.CustomContent custom component to present with text
 * @param {string} props.text text to present
 */
const UserModalContentTypographyGrid = ({ IconComponent, CustomContent, text, ...props }) => {
  return (
    <Grid container {...props}>
      <Grid item xs={2}>
        { IconComponent }
      </Grid>
      <Grid item xs={10}>
        {text
          ? <UserModalContentTypography text={text} />
          : CustomContent
        }
      </Grid>
    </Grid>
  );
};

export default UserModalContentTypographyGrid;

export {
  UserModalContentTypography
}

UserModalContentTypographyGrid.propTypes = {
  IconComponent: PropTypes.element.isRequired,
  CustomContent: PropTypes.element,
  text: PropTypes.string,
  props: PropTypes.arrayOf(PropTypes.any)
};

UserModalContentTypography.propTypes = {
  text: PropTypes.string.isRequired,
};