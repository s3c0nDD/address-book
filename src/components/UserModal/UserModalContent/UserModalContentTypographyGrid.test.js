import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import UserModalContentTypographyGrid, {
  UserModalContentTypography
} from './UserModalContentTypographyGrid';

describe('UserModalContentTypography', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it('shallowly renders and matches snapshot', () => {
    const wrapper = shallow(
      <UserModalContentTypography
        text="Hello world"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('UserModalContentTypographyGrid', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it('shallowly renders and matches snapshot', () => {
    const wrapper = shallow(
      <UserModalContentTypographyGrid
        text="Hello world"
        IconComponent={<div />}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('shallowly renders and matches snapshot (with CustomComponent prop)', () => {
    const wrapper = shallow(
      <UserModalContentTypographyGrid
        CustomContent={<div className="CustomComponent" />}
        IconComponent={<div className="IconComponent" />}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});