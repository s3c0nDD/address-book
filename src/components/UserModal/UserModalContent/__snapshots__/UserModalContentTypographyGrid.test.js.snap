// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModalContentTypography shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(Typography))
  color="textSecondary"
  component="p"
  variant="body2"
>
  Hello world
</WithStyles(ForwardRef(Typography))>
`;

exports[`UserModalContentTypographyGrid shallowly renders and matches snapshot (with CustomComponent prop) 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={2}
  >
    <div
      className="IconComponent"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={10}
  >
    <div
      className="CustomComponent"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`UserModalContentTypographyGrid shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={2}
  >
    <div />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={10}
  >
    <UserModalContentTypography
      text="Hello world"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
